packages/service-library/src/servicelib/aiohttp/tracing.py

```diff
@@ -61,6 +63,7 @@ def setup_tracing(
     tracer_provider.add_span_processor(BatchSpanProcessor(otlp_exporter))  # type: ignore[attr-defined] # https://github.com/open-telemetry/opentelemetry-python/issues/3713
     # Instrument aiohttp server and client
     AioHttpServerInstrumentor().instrument()
+    app.middlewares.append(middleware)
```

Comment:
this is the "magic line" that snaps everything into place

Reply:
Can you provide links where the middleware is supposeduly appended in the library? I am unable to find it.
Please check here on how middlewares are used: https://docs.aiohttp.org/en/stable/web_advanced.html#middlewares

Reply:
This 'kinda makes sense to me.
I would even put it at the end of all the instrumentation calls, but I think that the position does not matter.

Reply:
```
you can always print out what middlewares were added and DEBUG why your middlewares is wiped out. That is how one finds hidden bugs.
```

i dont know how to do that :--(

Reply:
*Update:* i have investigated this further and added a verbose comment about why this call is done, and why it is ok in our case to do so. Hope that suffices and makes it clear! :--)